Added initial VIVO configuration.

qjhart opened the pull request:
This update adds VIVO 1.10 as an additional Dockerfile.  In addition, it sets up a config/* directory so that one can update the configuration w/out rebuilding the Dockerfile (only for VIVO).  
There are a couple of gotchas:  One, the mariadb and solr files still need volumes to run fast, and somewhat arbitrarily,  I changed, the mariadb username, password.  Those should be moved to an env file, but I didn't do that, yet.

Commit:
Added initial VIVO configuration.

qjhart committed:
Merge pull request #1 from ucd-library/vivo-1.10

Added initial VIVO configuration.

awoods commented on vivo/runtime.properties.jr:
```diff
@@ -0,0 +1,342 @@
+# -----------------------------------------------------------------------------
```
Is there a reason for the inclusion of this file in the pull-request?

qjhart replied:
Nope. We'll remove.  We are also looking at a few other modifications to this, (mostly adding proper volumes) so you might delay, and wait for a few additional changes.

ht29 reviewed (commented):
I like these changes.. It surely brings us closer to have VIVO on docker hub.. Great going Quinn and UC Davis team to take this further along...

jrmerz committed:
added volumes, modified config for example, setup stdio logging

qjhart committed:
Updated README

Added information about running an example installation.

qjhart committed:
Create README.md

Small documentation to the vivo Dockerfile

qjhart commented:
We've updated this pull request, with a better example docker-compose file with proper volumes, updated vivo/Dockerfile to remove some cruft, some updates to the README, and at least acknowledged the current bugs (passwords, docker-compose up/down/up)

awoods commented on vivo/README.md:
```diff
@@ -0,0 +1,8 @@
+image is designed to not run without an external solr container, so any solr instances built with the vivo installaion
+are removed.
+
+Some example configuation files are included, but they can be overwritten, typepically by mounting a local configuration
```
Minor typo: "typepically" --> "typically"

awoods commented on vivo/README.md:
```diff
@@ -0,0 +1,8 @@
+# VIVO Dockerfile
+
+The VIVO [Dockerfile](vivo/Dockerfile) is based on a tomcat container and installs VIVO 1.10 using maven.  Note that this
+image is designed to not run without an external solr container, so any solr instances built with the vivo installaion
```
Rewording suggestion: 
```
...image is designed to run with an external solr container...
```

awoods commented on README.md:
```diff
@@ -1,29 +1,53 @@
+
+## VIVO Runtime Example
+
+The example [docker-compose.yml](docker-compose.yml) is a typical installation for trying out a simple VIVO installation in docker. This file starts three containers and uses the standard SDB system with the mariadb backend.  This example also shows how a local directory [example-config](example-config) is used to overwrite the default `runtime.properties` as installed by the Dockerfile.  Here th ...
```
It looks like `dockercompose.runtime.properties` is copied to the `WORKDIR`:
https://github.com/vivo-community/vivo-docker2/pull/2/files#diff-d37b8b0134a42eeaaf42ab4b3c3dfca4R31

Then the `example-config/runtime.properties` is mounted as a volume in the `WORKDIR`:
https://github.com/vivo-community/vivo-docker2/pull/2/files#diff-4e5e90c6228fd48698d074241c2ba760R24

If I understand correctly, the `dockercompose.runtime.properties` is overwritten by the `example-config/runtime.properties`. 

The purpose is to demonstrate that configuration files in a local `example-config` directory will take precedence over the configuration files that are included in this project?

qjhart replied:
Correct

awoods commented on Makefile:
```diff
@@ -0,0 +1,9 @@
+#! /usr/bin/make
```
At what point does the Makefile come into play?

qjhart replied:
It's a construct we use to remember simple tasks, like building the containers, or a nice alias for running them.  We can remove.

awoods reviewed (changes requested):
It is worth noting that the externalized Solr integration is not production-quality in VIVO 1.10.0. The in-progress work towards developing that functionality can be found in the branches:
https://github.com/vivo-project/Vitro/tree/sprint-search and
https://github.com/vivo-project/VIVO/tree/sprint-search

awoods commented:
@qjhart : any updates on your end?

Commit:
Requested vivo-community updates.

awoods commented on README.md:
```diff
@@ -1,29 +1,53 @@
+ If you get an error indicating that the database was not found, this could be due to a bug where the vivo instance is not waiting on the mariadb instance to initialize.  IF you have this error, try `docker-compose down; docker-compose up -d`.
+
+
+## VIVO Developement
```
typo: `Development`

Commit:
Spell check READMEs.
